Mage/src/main/java/mage/abilities/effects/AsThoughEffectImpl.java

```diff
@@ -84,34 +81,26 @@ public boolean isAbilityAppliedForAlternateCast(Card cardToCheck, Ability affect
      */
     protected boolean allowCardToPlayWithoutMana(UUID objectId, Ability source, UUID affectedControllerId, Game game) {
         Player player = game.getPlayer(affectedControllerId);
-        Card card = game.getCard(objectId);
-        if (card == null || player == null) {
+        if (player == null) {
             return false;
         }
-        if (!card.isLand(game)) {
-            if (card instanceof SplitCard) {
-                Card leftCard = ((SplitCard) card).getLeftHalfCard();
-                player.setCastSourceIdWithAlternateMana(leftCard.getId(), null, leftCard.getSpellAbility().getCosts());
-                Card rightCard = ((SplitCard) card).getRightHalfCard();
-                player.setCastSourceIdWithAlternateMana(rightCard.getId(), null, rightCard.getSpellAbility().getCosts());
-            } else if (card instanceof ModalDoubleFacesCard) {
-                Card leftCard = ((ModalDoubleFacesCard) card).getLeftHalfCard();
-                Card rightCard = ((ModalDoubleFacesCard) card).getRightHalfCard();
-                // some MDFC's are land.  IE: sea gate restoration
-                if (!leftCard.isLand(game)) {
-                    player.setCastSourceIdWithAlternateMana(leftCard.getId(), null, leftCard.getSpellAbility().getCosts());
-                }
-                if (!rightCard.isLand(game)) {
-                    player.setCastSourceIdWithAlternateMana(rightCard.getId(), null, rightCard.getSpellAbility().getCosts());
-                }
-            } else if (card instanceof AdventureCard) {
-                Card creatureCard = card.getMainCard();
-                Card spellCard = ((AdventureCard) card).getSpellCard();
-                player.setCastSourceIdWithAlternateMana(creatureCard.getId(), null, creatureCard.getSpellAbility().getCosts());
-                player.setCastSourceIdWithAlternateMana(spellCard.getId(), null, spellCard.getSpellAbility().getCosts());
+        if (game.getCard(objectId) == null) {
+            return false;
+        }
+        Card card = game.getCard(objectId).getMainCard();
```

Comment:
Be careful on changing. `getCard` and `getCard.getMainCard` are different calls. 

Effect can target:
- main;
- one part;
- all parts;

So there are possible use cases with one playable part.

Reply:
Yes. That's the part I'm currently a little unsure about. Given that this part of the code didn't complain during the tests, I don't think there is enough tests for this portion.

I'll run with coverage and take a look at adding more on it.

Reply:
There are some docs in `AsThoughEffect` interface for more info about use cases.